docs/scope-and-objectives.md

```diff
@@ -0,0 +1,144 @@
+
+## Summary
+
+We are building a set of Kubernetes cluster management APIs to enable common cluster lifecycle operations (create, scale, upgrade, destroy) across infrastructure providers.
+
+#### What is Cluster API?
+
+- An API definition for managing Kubernetes Clusters declaratively.
+- A reference implementation for the core logic.
+- An API definition for cluster lifecycle management action extension points
+
+## Glossary
+
+- __Cluster API__: Unless otherwise specified, this refers to the project as a whole.
+- __Infrastructure provider__: Refers to the source of computational resources (e.g. machines, networking, etc.). Examples for cloud include AWS, Azure, Google, etc.; for bare metal include VMware, MAAS, etc. When there is more than one way to obtain resources from the same infrastructure provider (e.g. EC2 vs. EKS) each way is referred to as a variant.
+Provider implementation: Existing Cluster API implementations consist of generic and infrastructure provider-specific logic. The infrastructure provider-specific logic is currently maintained in infrastructure provider repositories.
+- __Kubernetes-conformant__: A cluster that passes the Kubernetes conformance tests.
+- __Control plane__:
+   - __Self-provisioned__: A Kubernetes control plane consisting of pods or machines wholly managed by a single Cluster API deployment.
+   - __External__: A control plane offered and controlled by some system other than Cluster API (e.g., GKE, AKS, EKS, IKS).
+- __Core Add-ons__: Addons that are required to deploy a Kubernetes-conformant cluster: DNS, kube-proxy, CNI.
+- __Additional Add-ons__: Addons that are not required for a Kubernetes-conformant cluster (e.g. metrics/Heapster, Dashboard).
+- __Management cluster__: The cluster where one or more Cluster API providers run, and where resources (e.g. Machines) are stored.
+- __Workload cluster__: A cluster whose lifecycle is managed by the Management cluster.
+- __Operating system__: A generically understood combination of a kernel and system-level userspace interface, such as Linux or Windows, as opposed to a particular distribution.
+- __Manage a cluster__: create, scale, upgrade, destroy.
+
+## Motivation
+
+Kubernetes has a common set of APIs (see the [Kubernetes API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md)) to orchestrate containers regardless of deployment mechanism or cloud provider. Kubernetes also has APIs for handling some infrastructure, like load-balancers, ingress rules, or persistent volumes, but not for creating new machines.
+As a result, the deployment mechanisms that manage Kubernetes clusters each have unique APIs and implementations for how to handle lifecycle events like cluster creation or deletion, control plane upgrades, and node upgrades.
+
+### Goals
+
+- To manage the lifecycle (create, scale, upgrade, destroy) of Kubernetes-conformant clusters using a declarative API.
```

Comment:
This document refers to scaling, but doesn't suggest if it's horizontal scaling, vertical scaling, or both. Is calling it out relevant to the goal of this document?

Reply:
Might be worthwhile to add a definition for scaling to the glossary, we use it to refer to horizontal scaling, otherwise noted

Reply:
Added to the glossary